pkg/hive/hive.go

```diff
@@ -0,0 +1,275 @@
+// Copyright 2020 The Swarm Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package hive
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ethersphere/bee/pkg/hive/pb"
+	"github.com/ethersphere/bee/pkg/logging"
+	"github.com/ethersphere/bee/pkg/p2p"
+	"github.com/ethersphere/bee/pkg/p2p/protobuf"
+)
+
+const (
+	protocolName           = "hive"
+	initStreamName         = "hive_init"
```

Comment:
I think that prefix `hive_` is not needed for stream names as protocol name is in the stream id already, there will be no conflicts with streams that may be called the same in other protocols.